internal/databases/mongo/stream_fetch_handler.go

```
@@ -2,141 +2,57 @@ package mongo
+func fetchStreamSentinel(backup *internal.Backup) (StreamSentinelDto, error) {
 	sentinelDto := StreamSentinelDto{}
-	sentinelDtoData, err := backup.Backup.FetchSentinelData()
+	sentinelDtoData, err := backup.FetchSentinelData()
 	if err != nil {
 		return sentinelDto, errors.Wrap(err, "failed to fetch sentinel")
 	}
 	err = json.Unmarshal(sentinelDtoData, &sentinelDto)
 	return sentinelDto, errors.Wrap(err, "failed to unmarshal sentinel")
 }
 
-// TODO : unit tests
-func downloadAndDecompressStream(folder storage.Folder, fileName string) error {
-	baseBackupFolder := folder.GetSubFolder(utility.BaseBackupPath)
-	backup := Backup{internal.NewBackup(baseBackupFolder, fileName)}
-
-	streamSentinel, err := backup.FetchStreamSentinel()
-	if err != nil {
-		return err
-	}
-	oplogsAreDone := make(chan error)
-	go fetchOplogs(folder, streamSentinel.StartLocalTime, oplogsAreDone)
-
-	for _, decompressor := range compression.Decompressors {
-		archiveReader, exists, err := internal.TryDownloadWALFile(baseBackupFolder, getStreamName(backup.Name, decompressor.FileExtension()))
-		if err != nil {
-			return err
-		}
-		if !exists {
-			continue
-		}
-
-		err = internal.DecompressWALFile(&internal.EmptyWriteIgnorer{WriteCloser: os.Stdout}, archiveReader, decompressor)
-		if err != nil {
-			return err
-		}
-		utility.LoggedClose(os.Stdout, "")
-		tracelog.DebugLogger.Println("Waiting for oplogs")
-		err = <-oplogsAreDone
-		return err
-	}
-	return internal.NewArchiveNonExistenceError(fmt.Sprintf("Archive '%s' does not exist.\n", fileName))
-}
-
 func fetchOplogs(folder storage.Folder, startTime time.Time, oplogAreDone chan error) {
-	oplogFolder := folder.GetSubFolder(OplogPath)
-	endTS, oplogDstFolder := getOplogConfigs()
-	if oplogDstFolder == "" {
-		oplogAreDone <- errors.New("WALG_MONGO_OPLOG_DST is not configured")
+	endTS, oplogDstFolder, err := getOplogConfigs()
+	if err != nil {
+		oplogAreDone <- nil
```

Why are we suppressing error here?